src/Angor/Avalonia/Angor.Wallet/Infrastructure/Impl/WalletAppService.cs

```diff
@@ -0,0 +1,266 @@
+using Angor.Shared;
+using Angor.Shared.Models;
+using Angor.Shared.Services;
+using Angor.Wallet.Application;
+using Angor.Wallet.Domain;
+using Angor.Wallet.Infrastructure.Interfaces;
+using Blockcore.NBitcoin.BIP39;
+using CSharpFunctionalExtensions;
+
+namespace Angor.Wallet.Infrastructure.Impl;
+
+public class WalletAppService : IWalletAppService 
+{
+    public static readonly WalletId SingleWalletId = new(new Guid("8E3C5250-4E26-4A13-8075-0A189AEAF793"));
+    private const string SingleWalletName = "<default>";
+        
+    private readonly ISensitiveWalletDataProvider sensitiveWalletDataProvider;
+    private readonly IIndexerService _indexerService;
+    private readonly IWalletFactory walletFactory;
+    private readonly IWalletOperations _walletOperations;
+    
+    public WalletAppService(
+        ISensitiveWalletDataProvider sensitiveWalletDataProvider,
+        IIndexerService indexerService,
+        IWalletFactory walletFactory,
+        IWalletOperations walletOperations)
+    {
+        this.sensitiveWalletDataProvider = sensitiveWalletDataProvider;
+        _indexerService = indexerService;
+        this.walletFactory = walletFactory;
+        _walletOperations = walletOperations;
+    }
+
+    public async Task<Result<IEnumerable<WalletMetadata>>> GetMetadatas()
+    {
+        return new List<WalletMetadata>
+        {
+            new(SingleWalletName, SingleWalletId)
+        };
+    }
+
+    public async Task<Result<IEnumerable<BroadcastedTransaction>>> GetTransactions(WalletId walletId)
+    {
+        try
+        {
+            var sensitiveDataResult = await sensitiveWalletDataProvider.RequestSensitiveData(walletId);
+            if (sensitiveDataResult.IsFailure)
+            {
+                return Result.Failure<IEnumerable<BroadcastedTransaction>>(sensitiveDataResult.Error);
+            }
+
+            var (seed, passphrase) = sensitiveDataResult.Value;
+            
+            var walletWords = new WalletWords { Words = seed, Passphrase = passphrase };
+            var transactions = new List<BroadcastedTransaction>();
+                
+            var accountInfo = _walletOperations.BuildAccountInfoForWalletWords(walletWords);
+                
+            // Update existing UTXOs information
+            await _walletOperations.UpdateDataForExistingAddressesAsync(accountInfo);
+            await _walletOperations.UpdateAccountInfoWithNewAddressesAsync(accountInfo);
+
+            foreach (var address in accountInfo.AllAddresses())
```

Comment:
This can be optimised later on but looks correct

Reply:
We can implement any caching mechanism to handle those requests later :)